README.md

```diff
@@ -174,18 +174,18 @@ You can specify your own custom mapping function.
 
 ```lua
 local check_back_space = function()
```

Comment:
This isn't strictly related to this PR so apologies for noise, but I am trying to figure out at the moment what is this function for and how having this conditional in tab mapping changes things. Would appreciate if someone could shed some light here.

Reply:
It determines whether the cursor is within/in-front-of a word (returning `false`), or there is whitespace behind the caret (returning `true`). Based on this, a `<Tab>` mapping can determine whether it should indent or begin completion (whether to cycle completions is based on `pumvisible`, the first check). After all, without a word to complete we want to allow the `<Tab>` to passthrough or else we cannot indent anymore.

See the example in the README or my personal config, knowing the above should hopefully be enough to walk through line-by-line.

Reply:
Ah, so I was pretty close thinking about it. This then means that if I don't have tab set up to trigger completion in the first place this isn't really relevant to me at all, is it?

And great thanks for that explanation! I have seen this function around in compe as well, for quite a while and was always bit surprised that with or without it there wasn't any difference I could see.

Reply:
> This then means that if I don't have tab set up to trigger completion in the first place this isn't really relevant to me at all, is it?

Nope! If you're not using `<Tab>`, it won't matter.

> I have seen this function around in compe as well, for quite a while

The first time I saw it was [back in `coc.nvim`](https://github.com/neoclide/coc.nvim#example-vim-configuration). Blame shows [this commit](https://github.com/neoclide/coc.nvim/commit/8b55463159de9020bf61f8f0074ae7bad591d4de), though in that commit the function already existed. So, who knows how old it is or where it really comes from 🤔

Happy to help!

Reply:
Awesome, great thanks! You have helped me a lot here now!